controllers/front/unsubscribe.php

```diff
@@ -0,0 +1,122 @@
+        if (Configuration::get('CLERK_DATASYNC_SYNC_SUBSCRIBERS', $this->getLanguageId(), null, $this->getShopId()) == '0') {
+            http_response_code(403);
+            return array(
+                'success' => false,
+                'message' => 'Subscriber SYNC is disabled, enable it to handle unsubscribers with Clerk.io'
+            );
+        }
+
+        $email = strtolower(trim(pSQL(Tools::getValue('email'))));
+        if (empty($email)) {
+            http_response_code(400);
+            return array(
+                'success' => false,
+                'message' => 'No email provided'
+            );
+        }
+
+        $id_shop = (int) $this->getShopId();
+        $id_lang = (int) $this->getLanguageId();
+
+        // unsubscribe email
+        if (version_compare(_PS_VERSION_, '1.7.0', '>=')) {
+            $set_query = "UPDATE `" . _DB_PREFIX_ . "emailsubscription` e 
+                  LEFT JOIN `" . _DB_PREFIX_ . "lang` l ON l.id_lang = e.id_lang
+                  LEFT JOIN `" . _DB_PREFIX_ . "shop` s ON s.id_shop = e.id_shop
+                  SET e.active = '0'
+                  WHERE e.email = '$email' AND e.id_shop = '$id_shop' AND e.id_lang = '$id_lang'";
+
+            $set_result = Db::getInstance()->execute($set_query);
+
+        } elseif (version_compare(_PS_VERSION_, '1.6.2', '>=')) {
+            // ! not tested
+            $set_query = "UPDATE `" . _DB_PREFIX_ . "newsletter` e
+                LEFT JOIN `" . _DB_PREFIX_ . "shop` s ON s.id_shop = e.id_shop
+                SET active = '0'
+                WHERE e.email = '$email' AND id_shop = '$id_shop'";
+
+            $set_result = Db::getInstance()->execute($set_query);
+        }
+        // if the query fails
+        if (!$set_result) {
+            http_response_code(500);
+            return array(
+                'success' => false,
+                'message' => 'Failed to unsubscribe from Prestashop, did not unsubscribe from Clerk.io'
+            );
+        }
+        // send unsubscribe request to clerk https://api.clerk.io/v2/subscriber/unsubscribe
+        $url = 'https://api.clerk.io/v2/subscriber/unsubscribe';
+        $key = Configuration::get('CLERK_PUBLIC_KEY', $this->getLanguageId(), null, $this->getShopId());
+
+        $url_with_params = $url . "?email=" . urlencode($email) . "&key=" . urlencode($key);
+
+        // ? Which headers do we want to use
+        $options = array(
+            'http' => array(
+                'header' => "Content-type: application/x-www-form-urlencoded\r\n",
+                'method' => 'GET'
+            )
+        );
+        $context = stream_context_create($options);
+        $clerk_unsub_result = file_get_contents($url_with_params, false, $context);
+
+        // check if clerk unsub request was successful
+        $clerk_status = json_decode($clerk_unsub_result, true);
+        if ($clerk_status["status"] != "ok") {
+            http_response_code(500);
```

Comment:
Not 500, should be 422 I think. 500 is is the server fails to execute, this is more that the email was not found in clerk.

Reply:
Clerk will always return ok, only if the clerk server errors or you give it without email param, or an empty email param.
email=A gives a status "ok".

As it should not be possible to not give a valid email.
Perhaps it should just relay the response from clerk:
'clerk_message' => clerk_status

Reply:
I think this is a good idea, relay the Clerk msg.

Reply:
Should it resubscribe in prestashop?